File: app/src/main/java/com/edu/apnipadhai/utils/VideoInfo.java
package com.edu.apnipadhai.utils;

import android.graphics.Bitmap;

import androidx.annotation.NonNull;

public class VideoInfo {
    @NonNull
    private final String videoTitle;
    @NonNull
    private final String channelTitle;
    private final String url;
    @NonNull
    private final Bitmap thumbnail;

    VideoInfo(@NonNull String videoTitle, @NonNull String channelTitle, @NonNull Bitmap thumbnail, String url) {
        this.videoTitle = videoTitle;
        this.channelTitle = channelTitle;
        this.thumbnail = thumbnail;
        this.url = url;
    }



    @NonNull
    public String getVideoTitle() {
        return videoTitle;
    }

    public String getUrl() {
        return url;
    }

    @NonNull
    public String getChannelTitle() {
        return channelTitle;
    }

    @NonNull
    public Bitmap getThumbnail() {
        return thumbnail;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        VideoInfo videoInfo = (VideoInfo) o;

        if (!videoTitle.equals(videoInfo.videoTitle)) return false;
        if (!channelTitle.equals(videoInfo.channelTitle)) return false;
        return thumbnail.equals(videoInfo.thumbnail);
    }

    @Override
    public int hashCode() {
        int result = videoTitle.hashCode();
        result = 31 * result + channelTitle.hashCode();
        result = 31 * result + thumbnail.hashCode();
        return result;
    }
}

File: app/src/main/java/com/edu/apnipadhai/utils/VideoInfo.kt
package com.edu.apnipadhai.utils

import android.graphics.Bitmap

class VideoInfo internal constructor(
    val videoTitle: String,
    val channelTitle: String,
    val thumbnail: Bitmap,
    val url: String
) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val videoInfo = o as VideoInfo
        if (videoTitle != videoInfo.videoTitle) return false
        return if (channelTitle != videoInfo.channelTitle) false else thumbnail == videoInfo.thumbnail
    }

    override fun hashCode(): Int {
        var result = videoTitle.hashCode()
        result = 31 * result + channelTitle.hashCode()
        result = 31 * result + thumbnail.hashCode()
        return result
    }

}